packages/ws-server/src/webSocketServer.ts

```diff
@@ -63,10 +68,53 @@ function getMaxConnectionTTL() {
     return parseInt(process.env.WS_MAX_CONNECTION_TTL || '300000');
 }
 
-app.ws.use((ctx) => {
+// function to get maximum allowed connections from a single IP
+function getLimitPerIp() {
+    return parseInt(process.env.CONNECTION_LIMIT_PER_IP || '10');
+}
+
+async function handleConnectionClose(ctx) {
+    if (ctx.websocket.ipCounted) {
```

Comment:
Q: Should we use a similar approach tot he relay where we pulled out the rate limiting logic into a separate class instead of lumping it in the main server class?
I think this would be cleaner and easier to maintain.